### ISSUE TITLE
Speed up and decrease the download size of the front page

### ISSUE THREAD

**Issue by simonv3** [openfarmcc/OpenFarm#720]
I have recently acquired the privilege of living somewhere where the amount of data you use is carefully metered out and measured. I don't like downloading 14mb every time I access OpenFarm.

Most of that is the video, which is pretty, but do we really need it? I imagine that for a lot of people to whom OF is actually interesting they might be in similar situations.

An easy step is making sure the images on the front page aren't full size, but rather just thumbnails. I'm going to take care of that right now, but I think the main discussion here should be that video.

**Comment by roryaronson** [comment]
We can replace it with a nice photo of a garden?

We can also try compressing it more. I have a video background on the FarmBot homepage that is 30 seconds long and only 7mb. Also, because the OF video is only about half screen height, we could probably get a video down to about 2-4mb if we cropped it, compressed it more, and shortened it to 20 seconds.

Its about time we re-did it anyways, whether that is making another video or going with a photo.

**Comment by simonv3** [comment]
I think if we can find a nice hi-res photo that might be best. The current video is 7mb :/
Maybe there's something here? https://unsplash.com/

**Comment by roryaronson** [comment]
What if we pulled a random crop image? We could provide a link to the crop in the bottom right of the image.

**Comment by simonv3** [comment]
oooh that'd be neat.

**Comment by roryaronson** [comment]
Along those lines, the random image/crop could double as a search suggestion. The "What do you want to grow?" placeholder text could be moved above the search box, and the placeholder text could be the name of the random crop.

**Comment by simonv3** [comment]
Actually, on second thought, this could be a bad idea, since anyone can edit crop images.

http://www.bbc.com/news/world-europe-jersey-35860760

**Comment by roryaronson** [comment]
Bahahaha that's amazing! Well, instead we could have a handful (5-10?) images/crops that we hand select. I like random more because then it will be different every time and not get stale. But curated could be useful too for drawing attention to lesser known crops or really awesome images.

**Comment by simonv3** [comment]
Well, even if we hand select them, people can upload an image to any crop, which would still pose the same problem :/ Maybe if we list it a crop as "featured" only admin can edit it?

I've made some improvements to the actual front page by the way (issue at hand :P). I've removed the video for now and the total size is down to 2mb, half of which is now-more-compressed images.

**Comment by roryaronson** [comment]
True. With your suggestion we would be blocking people form editing in a time when that crop page would get an increase in traffic due to the feature, which isn't ideal for improving the page. Maybe there could be a way for admins to mark specific photos as featured?

**Comment by simonv3** [comment]
That's an idea - mark photos as featured instead of the crop itself, and then use the model to traverse back to the crop. That'd be cool. Separate issue?

**Comment by roryaronson** [comment]
At this point, yes! Cause looks like you fixed this one :) Everything loads
wayyyy faster now, thanks!

On Fri, Mar 25, 2016 at 12:09 PM Simon <notifications@github.com> wrote:

> That's an idea - mark photos as featured instead of the crop itself, and
> then use the model to traverse back to the crop. That'd be cool. Separate
> issue?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/openfarmcc/OpenFarm/issues/720#issuecomment-201434708>
>
-- 
Rory

**Comment by simonv3** [comment]
I was trying to add gzip to our css and js and such, but that broke *everything* because rails-assets is going through some changes #718.
